plugins/modules/xen_orchestra_instance.py

```diff
@@ -0,0 +1,335 @@
+      - UID of the target Virtual Machine. Required when O(state=absent), O(state=started), O(state=stopped) or
+        O(state=restarted).
+    type: str
+  label:
+    description: Label of the Virtual Machine to create, can be used when O(state=present).
+    type: str
+  description:
+    description: Description of the Virtual Machine to create, can be used when O(state=present).
+    type: str
+  template:
+    description:
+      - UID of a template to create Virtual Machine from.
+      - Muse be provided when O(state=present).
+    type: str
+  boot_after_create:
+    description: Boot Virtual Machine after creation, can be used when O(state=present).
+    type: bool
+    default: false
+requirements:
+  - websocket-client >= 1.0.0
+author:
+  - Samori Gorse (@shinuza) <samorigorse@gmail.com>
+seealso:
+  - name: Xen Orchestra documentation
+    description: Official documentation of Xen Orchestra CLI.
+    link: https://docs.xen-orchestra.com/architecture#xo-cli-cli
+"""
+
+
+EXAMPLES = r"""
+- name: Create a new virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: present
+    template: 355ee47d-ff4c-4924-3db2-fd86ae629676-a3d70e4d-c5ac-4dfb-999b-30a0a7efe546
+    label: This is a test from ansible
+    description: This is a test from ansible
+    boot_after_create: false
+
+- name: Start an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: started
+
+- name: Stop an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: stopped
+
+- name: Restart an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: stopped
+
+- name: Delete a virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: absent
+"""
+
+import json
+import ssl
+import traceback
+from time import sleep
+
+from ansible_collections.community.general.plugins.module_utils.version import LooseVersion
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+# 3rd party imports
+try:
+    HAS_WEBSOCKET = True
+    WEBSOCKET_IMP_ERR = None
+    import websocket
+    from websocket import create_connection
+
+    if LooseVersion(websocket.__version__) < LooseVersion('1.0.0'):
+        raise ImportError
+except ImportError:
+    WEBSOCKET_IMP_ERR = traceback.format_exc()
+    HAS_WEBSOCKET = False
+
+OBJECT_NOT_FOUND = 1
+VM_STATE_ERROR = 13
+
+
+class XenOrchestra(object):
+    CALL_TIMEOUT = 100
+    '''Number of 1/10ths of a second to wait before method call times out.'''
```

Comment:
I would suggest making this a parameter with the default value set to `100`, so that users can have control of that.

Reply:
Maybe changing the unit to `ms` - that is more usual in code than 1/10th of a second.